app/ui-react/packages/ui/src/Customization/apiClientConnectors/detail/ApiConnectorDetailBody.tsx

```diff
@@ -0,0 +1,167 @@
+import {
+  Button,
+  Card,
+  CardBody,
+  CardFooter,
+  CardHeader,
+  Title,
+} from '@patternfly/react-core';
+import * as React from 'react';
+import { PageSection } from '../../../Layout';
+import { ApiConnectorDetailConfig } from './ApiConnectorDetailConfig';
+import { ApiConnectorDetailConfigEdit } from './ApiConnectorDetailConfigEdit';
+
+export interface IApiConnectorDetailValues {
+  address?: string;
+  basePath?: string;
+  description?: string;
+  host?: string;
+  icon?: string;
+  name: string;
+}
+
+export interface IApiConnectorDetailBodyProps {
+  /**
+   * Properties
+   */
+  address?: string;
+  basePath?: string;
+  description?: string;
+  host?: string;
+  icon?: string;
+  name: string;
+
+  /**
+   * Property Labels
+   */
+  i18nLabelBaseUrl: string;
+  i18nLabelDescription: string;
+  i18nLabelHost: string;
+  i18nLabelName: string;
+
+  /**
+   * The localized text for buttons
+   */
+  i18nCancelLabel: string;
+  i18nEditLabel: string;
+  i18nSaveLabel: string;
+
+  i18nNameHelper: string;
+  i18nTitle: string;
+  i18nRequiredText: string;
+
+  /**
+   * The callback fired when submitting the form.
+   * @param e
+   */
+  handleSubmit: (e: any) => void;
+}
+
+export const ApiConnectorDetailBody: React.FunctionComponent<IApiConnectorDetailBodyProps> = ({
+  address,
+  basePath,
+  description,
+  handleSubmit,
+  host,
+  i18nLabelBaseUrl,
+  i18nLabelDescription,
+  i18nLabelHost,
+  i18nLabelName,
+  i18nCancelLabel,
+  i18nEditLabel,
+  i18nNameHelper,
+  i18nRequiredText,
+  i18nSaveLabel,
+  i18nTitle,
+  icon,
+  name,
+}) => {
+  const [configured, setConfigured] = React.useState<IApiConnectorDetailValues>(
+    { address, basePath, description, host, icon, name }
+  );
+  const [isEditing, setIsEditing] = React.useState(false);
+
+  const onCancelEditing = () => {
+    setConfigured({ address, basePath, description, host, icon, name });
+    setIsEditing(false);
+  };
+
+  const onHandleChange = (fieldName: string, value: string) => {
+    setConfigured({ ...configured, [fieldName]: value });
+  };
+
+  const onStartEditing = () => {
+    setIsEditing(true);
+  };
+
+  const onSubmit = () => {
+    setIsEditing(false);
+    handleSubmit(configured);
+  };
+
+  return (
+    <PageSection>
+      <div
+        className={'row row-cards-pf'}
```

Comment:
I think it'd be worth replacing this div with a layout component also.  I don't recall that `row` and `row-cards-pf` are used anymore.

Reply:
I think also if we can prefer the react components to using html + classnames we'll be better off.

Reply:
Turns out we didn't even need to add anything it after removing the `div`, just removed it and moved the `data-testid` over to the `PageSection` component